src/libraries/Common/src/System/IO/FileSystem.DirectoryCreation.Windows.cs

```diff
@@ -121,11 +130,12 @@ public static unsafe void CreateDirectory(string fullPath, byte[]? securityDescr
             // Handle CreateDirectory("X:\\") when X: doesn't exist. Similarly for n/w paths.
             if ((count == 0) && !somepathexists)
             {
-                string? root = Path.GetPathRoot(fullPath);
+                ReadOnlySpan<char> root = Path.GetPathRoot(fullPathSpan);
```

Comment:
why change this? you end up creating the string either way.

it might be nice if DirectoryExists overload existed that took a span, to save creating it, but not suggesting that's worth doing here. This is a rare code path, right?

Reply:
It's not worth it here in my opinion because this calls a WinAPI method (interop with spans doesn't work, does it?).